### DISCUSSION TITLE
The Tauri application build does not run unless the code is executed.

### DISCUSSION THREAD

**Discussion by anugrah478** [tauri-apps/tauri#8624] [General]
I seek to execute a Tauri-built application without access to its source code. My goal is to effortlessly run the application on any system simply by downloading and executing the provided executable, mirroring the functionality of the original code execution that triggers the Tauri build application.
![Screenshot (91)](https://github.com/tauri-apps/tauri/assets/154414495/9863c03c-e8b3-40b2-9a05-3e98d1d70fee)
![Screenshot (90)](https://github.com/tauri-apps/tauri/assets/154414495/39f41cbe-aec2-4c54-84d7-aa33f45cafda)

can anyone give a solution for it ?

**Comment by FabianLars** [comment]
Hi, to be honest i don't understand your question. So as i understand it the exe that the `tauri build` command produces is not sufficient for you, but in what way exactly? What is it missing?

**Reply by anugrah478** [reply]
exe build runs only when i run the code, i don't know the correct reason behind it (like i run the code with yarn start), i want to run the build without running the code.

**Reply by FabianLars** [reply]
Ah okay, now i understand which code you mean.
- Do you see any errors when you run the built exe?
- Are you opening the exe from the `src-tauri/target/release/` folder?
- If not, please rebuild your app with `yarn tauri build --debug` and open the exe from `src-tauri/target/debug/`. Then right click on the window -> Inspect to open the devtools window. Then check the devtools console and the native terminal for errors.

**Reply by anugrah478** [reply]
When I execute the yarn build command, no errors occur. I'm following the method you suggested earlier for opening the application. However, when I attempt to build with yarn tauri build --debug, I encounter a compiler error (the screenshot is provided below). Notably, there are no errors displayed in the console.
![Screenshot (92)](https://github.com/tauri-apps/tauri/assets/154414495/758ef1f2-a89e-45f4-b409-1c9a77f2d66c)
![Screenshot (93)](https://github.com/tauri-apps/tauri/assets/154414495/fa287402-c8a7-41b8-a649-e2dfa2472da0)

**Reply by FabianLars** [reply]
Yeah, you'll have to do what the error tells you to if you plan to share your app with others (or if you want to run `tauri build` in general). 

For testing you could literally change it to `com.tauri.dev2` but make sure you use something that's fairly unique to your app, following reverse domain notation is generally recommended (even if you don't own the domain). So for example `com.anugrah.tpmdda`

Then the build command should work. Btw running `yarn build` only builds your frontend, not the whole app so it makes sense that you still see the same errors. So yeah, change the identifier and follow my steps from the first message again :)

**Reply by anugrah478** [reply]
![image](https://github.com/tauri-apps/tauri/assets/154414495/b8b3b5f5-ae2c-4bc5-ae64-cb550d0b21b1)
In build app console is not working, is there any dependencies or anything to get console window ?

**Reply by FabianLars** [reply]
That's why i asked for `yarn build --debug` :) Builds created with the --debug flag will have the console enabled.

Alternatively you can add the `"devtools"` feature flag to the tauri dependency in Cargo.toml

**Comment by anugrah478** [comment]
![image](https://github.com/tauri-apps/tauri/assets/154414495/604af5c2-c910-431e-b04a-2600639b08ba)
Hello What adjustments can I make to fix this error that appears when I try to launch the current React application?

**Reply by FabianLars** [reply]
Access-Control-Allow-Origin is a header that has to be set by the server. So if you have any control over api.maridock.com i would suggest to fix it there.

If that's not feasabile you will have to use tauri's own http api https://tauri.app/v1/api/js/http because you can't disable cors for the browser's native http implementation and tauri's module is actually written in rust where we have full control and simply ignore cors.
